lib/token.cpp

```diff
@@ -272,6 +272,22 @@ void Token::takeData(Token *fromToken)
         _link->link(this);
 }
 
+void Token::deleteThisInPlace()
```

Comment:
I must say that I don't have a really good feeling about adding this `Token::deleteThisInPlace()` method. It is not safe for arbitrary use cases. It should only be called from `TokenList::deleteToken()`.

If it is possible to change your code in `Tokenizer::simplifyKnownVariables()` rather than adding `TokenList::deleteToken()` I would prefer that. For instance if the Tokens can be deleted in reverse order.